README.md

```diff
@@ -1,38 +1,123 @@
+  ``` implementation project(":offix-offline") ```
+
+**Sample project's build.gradle**
+
+```groovy
+    // Top-level build file where you can add configuration options common to all sub-projects/modules.
+    buildscript {
+        // ..other code..
+        dependencies {
+          classpath 'com.android.tools.build:gradle:3.4.1'
+          classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlin_version"
+          classpath 'com.apollographql.apollo:apollo-gradle-plugin:1.0.0'
+            // NOTE: Do not place your application dependencies here; they belong
+            // in the individual module build.gradle files
+        }
+    }
+```
+  
+**App's AndroidManifest.xml**
+
+ Add the permissions to access network state to determine if the device is offline and to access Internet while using the app.
+
+```xml
+    <uses-permission android:name="android.permission.ACCESS_NETWORK_STATE" />
+    <uses-permission android:name="android.permission.INTERNET"/>
+```  
 
-## Running the sample app
+- Generate your queries and mutations `.graphql files`, add the schema file for your app, build up the project to generate       the apollo generated code for the abbove files, and create an apollo-client in your application as mentioned in the apollo-   android documentation.
 
-1. Make a mutation by going offline.
-2. Then, come online and click on "FromDb" button and you will see the results.
-3. Also, refresh the application to see the updated cache.
+ **Consuming Code**
 
+  **Code for performing Mutations** 
+  
+  ```groovy
+    //Create an Object of mutation by passing in the build parameters according to the scehma.
+    val mutation = UpdateCurrentTaskMutation.builder().id(id).title(title).version(version).build()
+    
+    //Create a client which is an object of ApolloCall on which call would be made.
+     val client = Utils.getApolloClient(context)?.mutate(mutation)?.refetchQueries(apolloQueryWatcher?.operation()?.name())
+     
+    //Create a callback object of type ApolloCall.Callback<UpdateCurrentTaskMutation.Data>
+     val callback = object : ApolloCall.Callback<UpdateCurrentTaskMutation.Data>() {
+            override fun onFailure(e: ApolloException) {
+                e.printStackTrace()
+            }
+
+            override fun onResponse(response: Response<UpdateCurrentTaskMutation.Data>) {
+              //Perform UI Bindings here.
+            }
+        }
+        
+     /*Call the enqueue function on the instance of APolloClient and pass in two parameters here:
+       1. mutation object typecasted as mutation as 
+          com.apollographql.apollo.api.Mutation<Operation.Data, Any, Operation.Variables>                 
+       2. callback object typecasted as callback as ApolloCall.Callback<Any>
+     */  
+     Utils.getApolloClient(context)?.enqueue(
+            mutation as com.apollographql.apollo.api.Mutation<Operation.Data, Any, Operation.Variables>,
+            callback as ApolloCall.Callback<Any>
+        )
```

Comment:
@Lavanyagaur22 come to think of it, can we have a callback in our enqueue method that is called when all the mutations are completed so that the user can then update their UI?
Just a random thought.

Reply:
Not in the enqueue method, I guess. Because it should be provided as a parameter by the user then. User doesn't know when are the mutations getting completed.